Patches/PlayerControlPatch.cs

```diff
@@ -245,12 +245,12 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] PlayerC
 
         if (!targetRoleClass.OnCheckMurderAsTarget(killer, target))
         {
-            return false;
+            return targetRoleClass.OnCheckMurderAsTarget(killer, target);
         }
 
         if (!killerRoleClass.OnCheckMurderAsKiller(killer, target))
         {
-            return false;
+            return killerRoleClass.OnCheckMurderAsKiller(killer, target);
         }
 
```

Comment:
Same, no need check again

Reply:
the method won't run in an if statement, so "Witness Notify" for example wouldn't run. Or chatgpt told me that atleast.

Reply:
It will work, unless target returns `false`

![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/104814436/0dda41c7-c0cc-4e23-a795-87f2b0f8abf8)

Reply:
Although I need to create `ForceCheckMurderAsKiller` in the future, to prevent the target from running its check

Reply:
> It will work, unless target returns `false`
> 

won't it kill the target if I return true

Reply:
wait, or maybe are you thinking of a method that guards the target which should be put inside the method inside, if so I am not aware sorry

Reply:
If `OnCheckMurderAsTarget` or `OnCheckMurderAsKiller` return false, this means that all the code below will not be executed, respectively, if `OnCheckMurderAsKiller` returns true, then the target will be killed, but code in `OnCheckMurderAsKiller` will be executed

Reply:
Ohhh, so Ig I should make another method for that then? since it prevents stuff if not killed.

Reply:
But `Witness` can't kill anyone, so it should always `return false`

Reply:
Wait, you not add Witness here
![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/104814436/4a19830f-f680-4fb4-949a-bb7c82a9fb60)

Reply:
That's why this role doesn't work

Reply:
I didn't test it loll, I was thinking it all in theory. So it would execute even that it's run by an if statement?

Reply:
Yes, the code returning bool always executes the code until it reaches the code `return true/false;`

Reply:
damm, I've been lied by chatgpt thiss entire time, well maybe because I use the free version thoo

Reply:
If everything worked the way ChatGPT said, half the code would never work, lol

Reply:
haha true, I just come ask it sometimes for small questions, which don't leave much room for error 90% of the time